### PR TITLE
Well, overriding was done automatically by IJI

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -1,5 +1,7 @@
 package core.basesyntax;
 
+import java.util.Objects;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]: Let’s not use java.util.Objects

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -27,4 +29,25 @@ public Integer getLength() {
...
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return width.equals(rectangle.width)
+                &&
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]: there is no need to keep this operators at the separate line

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -27,4 +27,23 @@ public Integer getLength() {
...
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return width.equals(rectangle.width) &&
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        return width.equals(rectangle.width)
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 4]
```diff
@@ -27,4 +27,23 @@ public Integer getLength() {
...
+                length.equals(rectangle.length) &&
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
                && length.equals(rectangle.length) 
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 5]
```diff
@@ -27,4 +27,23 @@ public Integer getLength() {
...
+                color.equals(rectangle.color);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
                && color.equals(rectangle.color);
```

**Review by Tarabarnik — COMMENTED** [review]
try that way :)

File: src/main/java/core/basesyntax/Rectangle.java [hunk 6]
```diff
@@ -27,4 +27,23 @@ public Integer getLength() {
...
+        }
+        Rectangle rectangle = (Rectangle) o;
+        return width.equals(rectangle.width)
+                && length.equals(rectangle.length)
+                && color.equals(rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 * (width + length) + color.length();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]: this hashCode is very simple and may cause lots of collisions. Try to complicate it with hashcodes of Rectangle's fields. and don't forget, that they might be null :)

File: src/main/java/core/basesyntax/Rectangle.java [hunk 7]
```diff
@@ -27,4 +27,24 @@ public Integer getLength() {
...
+        Rectangle rectangle = (Rectangle) o;
+        return width.equals(rectangle.width)
+                && length.equals(rectangle.length)
+                && color.equals(rectangle.color);
+    }
+
+    @Override
+    public int hashCode() {
+        return 31 * ((width != null) ? width.hashCode() : 0)
+                + 31 * ((length != null) ? length.hashCode() : 0) + color.length();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]: color is also a field of Rectangle class. so if you ask its method without null check, you may get NullPointerException

Comment (IegorSoboliev) [reply src/main/java/core/basesyntax/Rectangle.java]: I had thought about it:(. Thank you!

**Pull request closed without merging**